app/contracts/projects/base_contract.rb

```diff
@@ -119,26 +116,14 @@ def with_unchanged_id
       model.id = project_id
     end
 
-    def with_active_assumed
-      active = model.active
-      model.active = true
-
-      yield
-    ensure
-      model.active = active
-    end
-
     def validate_changing_active
       return unless model.active_changed?
 
-      RequiresAdminGuard.validate_admin_only(user, errors)
+      contract_klass = model.being_archived? ? ArchiveContract : UnarchiveContract
```

Comment:
This is another argument why `ArchiveContract` and `UnarchiveContract` should not depend on `BaseContract` (see below).

If any of those are called from here, within the `BaseContract`, the rest of the validations will be run twice at least. I currently don't understand why it seems to not be causing an infinite call stack.

Reply:
I do not understand why the rest of the validations would run twice.

For instance the only things being checked by `Projects::ArchiveContract` are

```
    validate :validate_no_foreign_wp_references
    validate :validate_has_archive_project_permission
```

It won't run any other check.

Note that `Projects::ArchiveContract` inherits from `::BaseContract`, not `Projects::BaseContract`. That's maybe where the confusion is coming from. (and at first I forgot the `::`, and yes, the unit test was failing with a stack too deep error :) )

Reply:
> Note that Projects::ArchiveContract inherits from ::BaseContract, not Projects::BaseContract

That was my mistake, yes. Sorry for the confusion.